webapp/channels/src/components/search_results/index.tsx

```diff
@@ -80,8 +81,15 @@ function makeMapStateToProps() {
         const currentSearch = (getCurrentSearchForSearchTeam(state) as unknown as Record<string, any>) || {};
         const currentTeamName = getCurrentTeam(state)?.name ?? '';
 
+        if (ownProps.isPinnedPosts) {
+            results = results.sort((postA: Post|FileSearchResultItem, postB: Post|FileSearchResultItem) => postB.create_at - postA.create_at);
+        }
+
+        const addDateSeparatorsForSearchResults = makeAddDateSeparatorsForSearchResults();
```

Comment:
Similarly, could you move this out into the `makeMapStateToProps` so that we're using reusing the same instance of `addDateSeparatorsForSearchResults` instead of creating new ones each time?